bundles/org.openhab.ui.basic/web-src/smarthome.js

```diff
@@ -1089,29 +1089,28 @@
 				};
 			}
 			var
-				maxR = _t.image.clientWidth / 2,
-				offsetX = pos.x - maxR,
-				offsetY = pos.y - maxR,
-				r = (offsetY * offsetY) + (offsetX * offsetX);
+				maxRadius = _t.image.clientWidth / 2,
+				offsetX = pos.x - maxRadius,
+				offsetY = pos.y - maxRadius,
+				radius = Math.sqrt((offsetY * offsetY) + (offsetX * offsetX)),
+				angle = Math.atan2(offsetX, offsetY) / Math.PI / 2;
 
-			if (r > (maxR * maxR)) {
+			if (radius > maxRadius) {
 				var
-					ratio = 1 - Math.abs(maxR / Math.sqrt(r));
+					ratio = 1 - Math.abs(maxRadius / radius);
 
 				pos.x -= (offsetX * ratio);
 				pos.y -= (offsetY * ratio);
+				radius = maxRadius;
 			}
 
 			_t.handle.style.left = (pos.x / _t.image.clientWidth) * 100 + "%";
 			_t.handle.style.top = (pos.y / _t.image.clientWidth) * 100 + "%";
 
 			var
-				angle = offsetX >= 0 ?
-						(Math.PI * 2 - Math.atan(offsetY / offsetX) + Math.PI / 2) / (Math.PI * 2) :
-						(Math.PI * 2 - Math.atan(offsetY / offsetX) - Math.PI / 2) / (Math.PI * 2),
 				hsv = {
-					h: isNaN(angle) ? 0 : angle,
-					s: Math.sqrt(r) / maxR,
+					h: Math.min(1.0, Math.max(0.0, offsetX >= 0 ? angle : 1 + angle)),
```

Comment:
@lolodomo I think that strictly speaking the Math.min and Math.max are no longer needed, since (as far as I can tell), the angle code now always produces a value in the range (0.00 .. 1.00). However the min/max check certainly doesn't do any harm.

```
offsetX >= 0 ? angle : 1 + angle
```

Reply:
Are you sure that "1 + angle" can never be greater than 1 ?

Reply:
Pretty much ;) ..

reason is that angle is only > 0 if offsetX is > 0 (so in other words 1 + angle is < 1)

alternate (perhaps even better) code could be ..

```
// old
offsetX >= 0 ? angle : 1 + angle
..
// better
angle >= 0 ? angle : 1 + angle
```

Reply:
Done